library/Exceptions/NestedValidationException.php

```diff
@@ -22,76 +22,75 @@
 use function is_array;
 use function str_repeat;
 
+/**
+ * Exception for nested validations.
+ *
+ * This exception allows to have exceptions inside itself and providers methods
+ * to handle them and to retrieve nested messages based on itself and its
+ * children.
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Jonathan Stewmon <jstewmon@rmn.com>
+ * @author Wojciech Frącz <fraczwojciech@gmail.com>
+ */
 class NestedValidationException extends ValidationException implements IteratorAggregate
 {
     /**
      * @var SplObjectStorage
      */
-    private $exceptions = [];
+    private $exceptions;
 
     /**
-     * @param ValidationException $exception
+     * Returns the exceptions that are children of the exception.
      *
-     * @return self
+     * @return SplObjectStorage|ValidationException[]
      */
-    public function addRelated(ValidationException $exception)
+    public function getRelated(): SplObjectStorage
     {
-        $this->getRelated()->attach($exception);
+        if (!$this->exceptions instanceof SplObjectStorage) {
+            $this->exceptions = new SplObjectStorage();
+        }
 
-        return $this;
+        return $this->exceptions;
     }
 
     /**
-     * @return RecursiveIteratorIterator
+     * Adds a child to the exception.
+     *
+     * @param ValidationException $exception
+     *
+     * @return self
      */
-    private function getRecursiveIterator()
+    public function addRelated(ValidationException $exception)
     {
-        $exceptionIterator = new RecursiveExceptionIterator($this);
-        $recursiveIteratorIterator = new RecursiveIteratorIterator(
-            $exceptionIterator,
-            RecursiveIteratorIterator::SELF_FIRST
-        );
+        $this->getRelated()->attach($exception);
 
-        return $recursiveIteratorIterator;
+        return $this;
     }
 
     /**
-     * Returns weather an exception should be omitted or not.
+     * Adds children to the exception.
```

Comment:
```diff
- Adds children to the exception.
+ Adds children to exceptions.
```

Reply:
This method adds children to the `NestedValidationException`